src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -238,83 +324,44 @@ unsafe class EventPipeEventMetaData
         /// </summary>
         public EventPipeEventMetaData(PinnedStreamReader reader, int length, int fileFormatVersionNumber, int pointerSize, int processId)
         {
-            StreamLabel eventDataEnd = reader.Current.Add(length);
-
+            // Get the event record and fill in fields that we can without deserializing anything.  
             _eventRecord = (TraceEventNativeMethods.EVENT_RECORD*)Marshal.AllocHGlobal(sizeof(TraceEventNativeMethods.EVENT_RECORD));
             ClearMemory(_eventRecord, sizeof(TraceEventNativeMethods.EVENT_RECORD));
 
-            if (pointerSize == 4)
```

Comment:
This is another case where I prefer using if than ?:.  (because it does not fit on a line)     In general ?: is hard to read unless it is really short.   Yes, you are cloning the _eventRecord->EventHeader.Flags part but that is OK.


 
-                _eventRecord->EventHeader.Flags = TraceEventNativeMethods.EVENT_HEADER_FLAG_64_BIT_HEADER;

Reply:
I have simplified it to: `(ushort)(pointerSize * 8)`